src/api.rs

```diff
@@ -0,0 +1,169 @@
+                },
+                ChatEntry {
+                    id: 4,
+                    username: "Julian Montes de Oca".to_string(),
+                    latest_message: MessagePreview::Video,
+                    timestamp: "friday".to_string(),
+                },
+                ChatEntry {
+                    id: 5,
+                    username: "Edward Tan".to_string(),
+                    latest_message: MessagePreview::Text("thanks ed, see you there.".to_string()),
+                    timestamp: "thursday".to_string(),
+                },
+                ChatEntry {
+                    id: 6,
+                    username: "WeChat Team".to_string(),
+                    latest_message: MessagePreview::Text("Welcome to WeChat!".to_string()),
+                    timestamp: "18/07".to_string(),
+                },
+                ChatEntry {
+                    id: 7,
+                    username: "Andrew Lin".to_string(),
+                    latest_message: MessagePreview::Text(
+                        "Awesome, I'll make sure they know about it".to_string(),
+                    ),
+                    timestamp: "18/07".to_string(),
+                },
+                ChatEntry {
+                    id: 8,
+                    username: "Christian Huxley".to_string(),
+                    latest_message: MessagePreview::Image,
+                    timestamp: "15/07".to_string(),
+                },
+                ChatEntry {
+                    id: 9,
+                    username: "Ana Leddie".to_string(),
+                    latest_message: MessagePreview::Image,
+                    timestamp: "14/07".to_string(),
+                },
+                ChatEntry {
+                    id: 10,
+                    username: "Adam Adler".to_string(),
+                    latest_message: MessagePreview::Video,
+                    timestamp: "10/07".to_string(),
+                },
+                ChatEntry {
+                    id: 11,
+                    username: "Gabriel Hayes".to_string(),
+                    latest_message: MessagePreview::Text("wow I haven't seen that".to_string()),
+                    timestamp: "10/07".to_string(),
+                },
+                ChatEntry {
+                    id: 12,
+                    username: "Eric Ford".to_string(),
+                    latest_message: MessagePreview::Text("Nice to see you here!".to_string()),
+                    timestamp: "10/07".to_string(),
+                },
+            ],
+        }
+    }
+
+    pub fn get_all_chats(&self) -> &Vec<ChatEntry> {
+        &self.chats
+    }
+
+    pub fn get_chat(&self, chat_id: u64) -> Option<&ChatEntry> {
+        self.chats.iter().find(|m| m.id == chat_id)
+    }
+
+    pub fn get_messages_by_chat_id(&self, chat_id: u64) -> Vec<MessageEntry> {
+        self.messages
+            .iter()
+            .filter(|m| m.chat_id == chat_id)
+            .cloned()
+            .collect()
+    }
```

Comment:
I think for benchmarking we should make this into a hash or something better performing. But I'm ok with moving forward with this now